packages/vuetify/src/composables/dimensions.ts

```diff
@@ -1,63 +1,50 @@
+// Utilities
 import { computed } from 'vue'
-
-// Types
-import type { Prop } from 'vue'
-
-// Utils
-import { convertToUnit, keys } from '@/util/helpers'
+import { convertToUnit } from '@/util/helpers'
 import propsFactory from '@/util/propsFactory'
 
-// Props
-const allDimensionsProps = {
-  height: {
-    type: [Number, String],
-  },
-  maxHeight: {
-    type: [Number, String],
-  },
-  maxWidth: {
-    type: [Number, String],
-  },
-  minHeight: {
-    type: [Number, String],
-  },
-  minWidth: {
-    type: [Number, String],
-  },
-  width: {
-    type: [Number, String],
-  },
-}
-
 // Types
-type PropValue = string | number | null | undefined
-type PropNames = keyof typeof allDimensionsProps
-
-// Effect
-export function dimensionsFactory<S extends PropNames> (...possibleProps: S[]) {
-  const selectedProps = possibleProps.length ? possibleProps : keys(allDimensionsProps) as S[]
-
-  const makeDimensionsProps = propsFactory(selectedProps.reduce((obj, prop) => {
-    obj[prop] = allDimensionsProps[prop]
-    return obj
-  }, {} as Record<S, Prop<PropValue>>))
-
-  const useDimensions = (props: Partial<Record<S, PropValue>>) => {
-    const dimensions = computed(() => {
-      return selectedProps.reduce((obj, key) => {
-        const value: PropValue = props[key]
-        if (value) {
-          obj.style[key] = convertToUnit(value)
-        }
-        return obj
-      }, { style: {} } as { style: Record<S, string | undefined> })
-    })
-
-    return { dimensions }
-  }
+export type DimensionKey = keyof DimensionProps
+export type DimensionStyles = Partial<Record<DimensionKey, string>>
+
+export interface DimensionProps {
+  height?: number | string
+  maxHeight?: number | string
+  maxWidth?: number | string
+  minHeight?: number | string
+  minWidth?: number | string
+  width?: number | string
+}
 
-  return {
-    makeDimensionsProps,
-    useDimensions,
-  }
+// Composables
+export const makeDimensionProps = propsFactory({
```

Comment:
```patch
- export const makeDimensionProps = propsFactory({
+ export const makeDimensionProps = propsFactory<PropOptions<DimensionPropNumberString>>({
```

Reply:
> Otherwise each prop seem to infer into <any>

Make sure you've run `yarn`, `propsFactory` already infers types correctly without manual annotation.